jsapp/js/account/plans/planContainer.component.tsx

```diff
@@ -151,67 +203,140 @@ export const PlanContainer = ({
         <div className={styles.currentPlan}>{t('Your plan')}</div>
-      ) : (
-        <div />
-      )}
+      ) : isSubscribedProduct(product, submissionQuantity) ? (
+        <div className={styles.currentPlan}>{t('Your plan')}</div>
+      ) : null}
       <div
         className={classnames({
-          [styles.planContainerWithBadge]: isSubscribedProduct(price),
+          [styles.planContainerWithBadge]: isSubscribedProduct(
+            product,
+            submissionQuantity
+          ),
           [styles.planContainer]: true,
         })}
       >
         <h1 className={styles.priceName}>
-          {price.prices?.unit_amount
-            ? price.name
-            : freeTierOverride?.name || price.name}
+          {product.price?.unit_amount
+            ? product.name
+            : freeTierOverride?.name || product.name}
         </h1>
-        <PriceDisplay
-          price={price.prices}
-          submissionQuantity={submissionQuantity}
-        />
-        {price.prices.transform_quantity && (
-          <TextBox
-            label={t('Total Submissions per Month')}
-            errors={error.length ? error : false}
-            type={'number'}
-            onChange={onSubmissionsChange}
-            value={submissionQuantity.toString()}
-          />
-        )}
+        <div className={styles.priceTitle}>{displayPrice}</div>
         <ul className={styles.featureContainer}>
-          {Object.keys(price.metadata).map(
+          {product.price.transform_quantity && (
+            <>
+              <li className={styles.selectableFeature}>
+                <Icon name='check' size='m' color='teal' />
+                <KoboSelect
+                  name={t('Total Submissions per Month')}
+                  options={submissionOptions}
+                  size={'s'}
+                  type={'outline'}
+                  onChange={onSubmissionsChange}
+                  selectedOption={submissionQuantity.toString()}
+                />
+              </li>
+              <li>
+                <div className={styles.iconContainer}>
+                  <Icon
+                    name='check'
+                    size='m'
+                    color={product.price.unit_amount ? 'teal' : 'storm'}
+                  />
+                </div>
+                {t('##asr_minutes## minutes of automated transcription /month')
+                  .replace(
+                    '##asr_minutes##',
+                    (
+                      (adjustedQuantity *
+                        (parseInt(product.metadata?.nlp_seconds_limit || '0') ||
+                          parseInt(
+                            product.price.metadata?.nlp_seconds_limit || '0'
+                          ))) /
+                      60
+                    ).toLocaleString()
+                  )
+                  .replace(
+                    '##plan_interval##',
+                    product.price.recurring!.interval
+                  )}
```

Comment:
All these functions deep in the jsx gets hard to read. Could it just be a function?